app/scripts/lib/setupSentry.js

```diff
@@ -146,31 +154,131 @@ export default function setupSentry({ release, getState }) {
+ * from its `data` property, it particular those possibly found at
+ * data.from, data.to and data.url
+ *
+ * @param {object} breadcrumb - A Sentry breadcrumb object: https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
+ * @returns {object} A modified Sentry breadcrumb object.
+ */
+export function removeUrlsFromBreadCrumb(breadcrumb) {
+  if (breadcrumb?.data?.url) {
+    breadcrumb.data.url = hideUrlIfNotInternal(breadcrumb.data.url);
+  }
+  if (breadcrumb?.data?.to) {
+    breadcrumb.data.to = hideUrlIfNotInternal(breadcrumb.data.to);
+  }
+  if (breadcrumb?.data?.from) {
+    breadcrumb.data.from = hideUrlIfNotInternal(breadcrumb.data.from);
+  }
+  return breadcrumb;
+}
+
+/**
+ * Receives a Sentry event object and modifies it before the
+ * error is sent to Sentry. Modifications include both sanitization
+ * of data via helper methods and addition of state data from the
+ * return value of the second parameter passed to the function.
+ *
+ * @param {object} report - A Sentry event object: https://develop.sentry.dev/sdk/event-payloads/
+ * @param {Function} getState - A function that should return an object representing some amount
+ * of app state that we wish to submit with our error reports
+ * @returns {object} A modified Sentry event object.
+ */
+export function rewriteReport(report, getState) {
+  try {
+    // simplify certain complex error messages (e.g. Ethjs)
+    simplifyErrorMessages(report);
+    // remove urls from error message
+    sanitizeUrlsFromErrorMessages(report);
+    // Remove evm addresses from error message.
+    // Note that this is redundent with data scrubbing we do within our sentry dashboard,
+    // but putting the code here as well gives public visibility to how we are handling
+    // privacy with respect to sentry.
+    sanitizeAddressesFromErrorMessages(report);
+    // modify report urls
+    rewriteReportUrls(report);
+    // append app state
+    if (getState) {
+      const appState = getState();
+      if (!report.extra) {
+        report.extra = {};
       }
-    } catch (err) {
-      console.warn(err);
+      report.extra.appState = appState;
     }
-    return report;
+  } catch (err) {
+    console.warn(err);
   }
+  return report;
+}
 
-  return Sentry;
+/**
+ * Receives a Sentry event object and modifies it so that urls are removed from any of its
+ * error messages.
+ *
+ * @param {object} report - the report to modify
+ */
+function sanitizeUrlsFromErrorMessages(report) {
+  rewriteErrorMessages(report, (errorMessage) => {
+    let newErrorMessage = errorMessage;
+    const re = /(([-.+a-zA-Z]+:\/\/)|(www\.))\S+[@:.]\S+/gu;
+    const urlsInMessage = newErrorMessage.match(re) || [];
+    urlsInMessage.forEach((url) => {
+      try {
+        const urlObj = new URL(url);
+        const { hostname } = urlObj;
+        if (
+          !Object.values(ERROR_URL_ALLOWLIST).some(
+            (allowedUrl) =>
+              hostname === allowedUrl || hostname.endsWith(`.${allowedUrl}`),
```

Comment:
Nit: maybe more clear to call this variable `allowedHostname`.
```suggestion
            (allowedHostname) =>
              hostname === allowedUrl || hostname.endsWith(`.${allowedHostname}`),
```